### DISCUSSION TITLE
implementing Trilium Rocks share theme

### DISCUSSION THREAD

**Discussion by maphew** [TriliumNext/Trilium#6012] [Ideas] [answered]
I'm trying to implement the https://trilium.rocks theme. 

* Trilium Rocks theme setup guide - https://trilium.rocks/xecUJ8eL3pvM
* Code - https://github.com/zerebos/trilium.rocks

I have the css part working but not the javascript parts. What is the "Embedded Javascript" referred to in the guide? I don't see this in any of the option pages for TriliumNext (v0.91.2-beta)

>As for the note itself, it is yet another code note but this time of the type `Embedded JavaScript`. If you don't see this in your dropdown list when creating the note, you'll have to go enable it in Trilium's settings.

cc @zerebos

**Comment by zerebos** [comment]
Admittedly I've not been following TriliumNext development as close as I would have liked for a huge number of reasons honestly. But I can explain from my previous knowledge (this will be similar to what I posted in Discord).

Embedded JavaScript is a templating language that Trilium (and TriliumNext) use for generating share pages. It was part of the code note types that had to be enabled in settings at least for Trilium not sure if that has been removed in TriliumNext. But there should be a list of code "languages" you can enable in settings to add to the code note dropdown. `Embedded JavaScript` _should_ be an option there.

**Reply by maphew** [reply]
Aha! 'Embedded Javascript' is one of the file types listed under _Options >> Code Notes_. I had unchecked most of the types in order to shorten the amount of drop down scrolling needed in daily use. Re-enabling that allowed me to select embedded javascript for the note types as per the guide. 

However apparently that wasn't the actual issue as my share pages still are not showing the left hand navigation items (c.f. https://www.maphew.com/share/OnlyOne)

**Reply by maphew** [reply]
Looking at the template pages I see 4 references to what look like hardcoded noteId:
<img width="400" alt="2024-12-26 15_38_55-scripts js - trilium rocks  Codespaces_ effective palm-tree  - Visual Studio Cod" src="https://github.com/user-attachments/assets/ee21ece9-e79d-4d9b-9041-01aafdc8f09b" />

**Reply by zerebos** [reply]
Yes see this little section... https://trilium.rocks/xecUJ8eL3pvM#code-highlighting if you want code highlighting then you need to make that modification yourself until I get around to making it a bundle of some kind

**Reply by maphew** [reply]
yay! I sorted out my problem: I'd put the contents of `page.ejs` as a child node to Template note instead of as the content of the Template page.

As of TriliumNext v0.90.12 code highlighting is now a core feature so I can live with that part of the template not working as designed. Thanks for your help.

**Comment by maphew** [comment, marked as answer]
For anyone coming across this thread later, start with https://github.com/perfectra1n/trilium.rocks/releases/tag/1.0.0 from @perfectra1n and you don't need to do the npm parts described in the guide, and will sidestep the mistake I made by not putting the page.ejs content in the right note. 

For the step [Template Notes](https://trilium.rocks/xecUJ8eL3pvM#template-notes) section, 'Embedded Javascript' is one of the file types listed under _Options >> Code Notes_.

**Reply by perfectra1n** [reply]
I updated the release notes to be more straightforward, hopefully having to avoid looking at all those steps together :)

**Reply by zerebos** [reply]
I'll have to try and integrate something similar into mine

**Comment by maphew** [comment]
Fwiw, I've got TR theme working on my site and have started down what will proabably be a long and winding path to adapting it to my tastes. Thanks @zerebos and @perfectra1n for getting me off the starting block! Current state with a wee bit of explanation at https://www.maphew.com/share/trilium-share-theme-customization
